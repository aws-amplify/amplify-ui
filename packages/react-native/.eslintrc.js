module.exports = {
  ignorePatterns: ['dist', '.eslintrc.js', '*.config.js', 'jest.setup.js'],
  extends: ['@aws-amplify/amplify-ui/react'],
  overrides: [
    {
      extends: ['@aws-amplify/amplify-ui/jest'],
      files: ['**/__mocks__/**', '**/__tests__/**'],
    },
  ],
  // point to local tsconfig
  parserOptions: { project: ['tsconfig.json'], tsconfigRootDir: __dirname },
};
