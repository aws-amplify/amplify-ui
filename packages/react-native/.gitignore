dist
.watchmanconfig

# ignore jest generated coverage directory
coverage 
