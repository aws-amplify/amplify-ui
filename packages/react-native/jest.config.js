module.exports = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react$': '<rootDir>/node_modules/react',
    '^react-native$': '<rootDir>/node_modules/react-native',
  },
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.js'],
};
