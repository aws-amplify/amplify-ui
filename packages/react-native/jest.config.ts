import { Config } from 'jest';

const config: Config = {
  preset: 'react-native',
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  collectCoverage: true,
  collectCoverageFrom: [
    '<rootDir>/src/**/*.{js,jsx,ts,tsx}',
    '!<rootDir>/src/**/*{c,C}onstants.ts',
  ],
  moduleNameMapper: {
    '^react-native$': '<rootDir>/node_modules/react-native',
    '^uuid$': '<rootDir>/../../node_modules/uuid',
  },
  modulePaths: ['<rootDir>/node_modules/'],
  coverageThreshold: {
    global: {
      branches: 90,
      functions: 90,
      lines: 90,
      statements: 90,
    },
  },
  setupFiles: ['<rootDir>/jest.setup.ts'],
};

export default config;
