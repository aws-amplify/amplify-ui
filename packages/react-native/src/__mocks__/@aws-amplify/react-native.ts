export default {
  loadUrlPolyfill: jest.fn(),
};
