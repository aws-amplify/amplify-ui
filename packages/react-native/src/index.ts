export {
  Authenticator,
  AuthenticatorProps,
  useAuthenticator,
  UseAuthenticator,
  withAuthenticator,
  WithAuthenticatorOptions,
} from './Authenticator';
export * from './InAppMessaging';
export {
  Theme,
  ThemeProvider,
  defaultDarkModeOverride,
  defaultTheme,
  useTheme,
} from './theme';
