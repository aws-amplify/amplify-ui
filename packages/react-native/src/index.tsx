import React from 'react';
import { Text, View } from 'react-native';

export function HelloWorld(): JSX.Element {
  return (
    <View>
      <Text>Hello World!</Text>
    </View>
  );
}
