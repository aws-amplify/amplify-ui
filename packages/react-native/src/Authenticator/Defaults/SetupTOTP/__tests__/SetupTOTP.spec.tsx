import React from 'react';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import { authenticatorTextUtil } from '@aws-amplify/ui';

import { SetupTOTP } from '..';

const code = {
  name: 'code',
  label: 'Code',
  placeholder: 'Code',
  type: 'default' as const,
};
const fields = [code];
const toSignIn = jest.fn();

const props = {
  error: null as unknown as string,
  fields,
  Footer: SetupTOTP.Footer,
  FormFields: SetupTOTP.FormFields,
  Header: SetupTOTP.Header,
  handleBlur: jest.fn(),
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  hasValidationErrors: false,
  isPending: false,
  toSignIn,
  totpSecretCode: "Let's keep it hush hush",
};

const {
  getBackToSignInText,
  getConfirmingText,
  getConfirmText,
  getSetupTOTPText,
} = authenticatorTextUtil;

describe('SetupTOTP', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders as expected', () => {
    const { toJSON, getAllByRole, getByText } = render(
      <SetupTOTP {...props} />
    );

    expect(toJSON()).toMatchSnapshot();

    expect(getAllByRole('header')).toBeDefined();
    expect(getByText(getSetupTOTPText())).toBeDefined();
    expect(getByText(getConfirmText())).toBeDefined();
  });

  it('renders an error message', () => {
    const errorMessage = 'Test error message';
    const { toJSON, getByText } = render(
      <SetupTOTP {...props} error={errorMessage} />
    );

    expect(toJSON()).toMatchSnapshot();
    expect(getByText(errorMessage)).toBeDefined();
  });

  it('calls toSignIn an secondary button press', () => {
    const { getByText } = render(<SetupTOTP {...props} />);

    const secondaryButton = getByText(getBackToSignInText());

    expect(secondaryButton).toBeDefined();

    fireEvent(secondaryButton, 'press');

    expect(toSignIn).toHaveBeenCalledTimes(1);
  });

  it('shows the correct submit button based on isPending', async () => {
    const { queryByText } = render(<SetupTOTP {...props} isPending />);

    await waitFor(() => {
      expect(queryByText(getConfirmingText())).toBeDefined();
      expect(queryByText(getConfirmText())).toBe(null);
    });
  });
});
