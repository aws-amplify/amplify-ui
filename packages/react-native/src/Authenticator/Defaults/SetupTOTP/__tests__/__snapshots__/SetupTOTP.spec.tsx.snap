// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupTOTP renders an error message 1`] = `
Array [
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "hsl(210, 50%, 10%)",
        },
        Object {
          "fontSize": 24,
          "fontWeight": "300",
          "lineHeight": 36,
        },
        undefined,
      ]
    }
  >
    Setup TOTP
  </Text>,
  <View>
    <View
      accessibilityLabel="Code"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 21,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            Array [
              Object {
                "color": "hsl(210, 25%, 25%)",
                "lineHeight": 21,
                "padding": 4,
              },
              undefined,
            ],
          ]
        }
      >
        Code
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "hsl(210, 8%, 55%)",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 8,
            "paddingVertical": 12,
          }
        }
        testID="amplify__text-field__input-container"
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Code"
          placeholderTextColor="hsl(210, 10%, 40%)"
          selectionColor="hsl(210, 50%, 10%)"
          style={
            Array [
              Object {
                "color": "hsl(210, 50%, 10%)",
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>,
  <View
    accessibilityRole="alert"
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "hsl(0, 75%, 85%)",
          "flexDirection": "row",
          "paddingHorizontal": 8,
          "paddingVertical": 32,
        },
        undefined,
      ]
    }
  >
    <Image
      accessibilityRole="image"
      source={
        Object {
          "testUri": "../../../src/assets/icons/error.png",
        }
      }
      style={
        Array [
          Object {
            "height": 20,
            "resizeMode": "contain",
            "tintColor": undefined,
            "width": 20,
          },
          Object {
            "margin": 8,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "hsl(0, 100%, 20%)",
            "flex": 1,
            "paddingHorizontal": 8,
          },
          undefined,
        ]
      }
    >
      Test error message
    </Text>
  </View>,
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 4,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        Object {
          "backgroundColor": "hsl(190, 95%, 30%)",
        },
        null,
        null,
        Object {
          "marginVertical": 8,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "hsl(0, 0%, 100%)",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Confirm
    </Text>
  </View>,
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 4,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        Object {},
        null,
        null,
        Object {
          "marginVertical": 8,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "hsl(190, 95%, 30%)",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Back to Sign In
    </Text>
  </View>,
]
`;

exports[`SetupTOTP renders as expected 1`] = `
Array [
  <Text
    accessibilityRole="header"
    style={
      Array [
        Object {
          "color": "hsl(210, 50%, 10%)",
        },
        Object {
          "fontSize": 24,
          "fontWeight": "300",
          "lineHeight": 36,
        },
        undefined,
      ]
    }
  >
    Setup TOTP
  </Text>,
  <View>
    <View
      accessibilityLabel="Code"
      accessibilityState={
        Object {
          "disabled": false,
        }
      }
      accessible={true}
      style={
        Array [
          Object {
            "alignItems": "flex-start",
          },
          undefined,
        ]
      }
    >
      <Text
        accessibilityRole="text"
        style={
          Array [
            Object {
              "fontSize": 14,
              "fontWeight": "400",
              "lineHeight": 21,
            },
            Object {
              "color": "hsl(210, 50%, 10%)",
            },
            Array [
              Object {
                "color": "hsl(210, 25%, 25%)",
                "lineHeight": 21,
                "padding": 4,
              },
              undefined,
            ],
          ]
        }
      >
        Code
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "borderColor": "hsl(210, 8%, 55%)",
            "borderRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
            "paddingHorizontal": 8,
            "paddingVertical": 12,
          }
        }
        testID="amplify__text-field__input-container"
      >
        <TextInput
          accessible={true}
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Code"
          placeholderTextColor="hsl(210, 10%, 40%)"
          selectionColor="hsl(210, 50%, 10%)"
          style={
            Array [
              Object {
                "color": "hsl(210, 50%, 10%)",
                "flexGrow": 1,
                "fontSize": 16,
              },
              undefined,
            ]
          }
        />
      </View>
    </View>
  </View>,
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 4,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        Object {
          "backgroundColor": "hsl(190, 95%, 30%)",
        },
        null,
        null,
        Object {
          "marginVertical": 8,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "hsl(0, 0%, 100%)",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Confirm
    </Text>
  </View>,
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderRadius": 4,
          "justifyContent": "center",
          "paddingHorizontal": 16,
          "paddingVertical": 12,
        },
        Object {},
        null,
        null,
        Object {
          "marginVertical": 8,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "hsl(190, 95%, 30%)",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Back to Sign In
    </Text>
  </View>,
]
`;
